presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SymbolMapper.java

```diff
@@ -55,10 +55,15 @@
 {
     private final Map<String, String> mapping;
 
-    public SymbolMapper(Map<Symbol, Symbol> mapping)
+    public SymbolMapper(Map<String, String> mapping)
```

Comment:
This probably a dumb question (or maybe the answer will come in later commits), but why not use a `VariableReferenceExpression` to `VariableReferenceExpression` map?

Reply:
Eventually this will be changed to `VariableMapper` and maps from variable to variable. But for most part of this PR, we have both symbols and variables, so `SymbolMapper` needs to work for both.